content/courses/native-onchain-development/paging-ordering-filtering-data-frontend.md

```diff
@@ -151,17 +166,26 @@ async function fetchMatchingContactAccounts(
   connection: web3.Connection,
   search: string,
 ): Promise<(web3.AccountInfo<Buffer> | null)[]> {
-  const accounts = await connection.getProgramAccounts(programId, {
-    dataSlice: { offset: 0, length: 0 },
-    filters: [
-      {
-        memcmp: {
-          offset: 13,
-          bytes: bs58.encode(Buffer.from(search)),
-        },
-      },
-    ],
-  });
+  const accounts = (await connection.getProgramAccounts(
+    new PublicKey(MOVIE_REVIEW_PROGRAM_ID),
+    {
+      dataSlice: { offset: 2, length: 18 },
+      filters:
+        search === ""
+          ? []
+          : [
+              {
+                memcmp: {
+                  offset: 6,
```

Comment:
Why 6? Use a variable or at least add a comment.

Reply:
comment added

Reply:
Where? I can't see this. It still shows `offset: 6,` and the code isn't marked as outdated.